pages/authors/[author].tsx

```diff
@@ -0,0 +1,36 @@
+import { GetStaticPaths, GetStaticProps } from 'next'
+
+import Author, { Props as AuthorProps } from '../../components/Author'
+import getQueryParameter from '../../util/getQueryParameters'
+import loadAllRecords from '../../util/loadAllRecords'
+import loadAuthorCollections from '../../util/loadAuthorCollections'
+import omitUndefinedFields from '../../util/omitUndefinedFields'
+
+export const getStaticPaths: GetStaticPaths = async () => {
+    const authorCollection =  await loadAuthorCollections()
+    const authorIds = authorCollection.authors.map(author => author.id)
+    const paths = authorIds.map(authorId => `/authors/${authorId}`)
+    return {
+        paths,
+        fallback: false,
+    }
+}
+
+export const getStaticProps: GetStaticProps<AuthorProps> = async context => {
+    const authorId = getQueryParameter(context.params, 'author')
+    const authorCollection =  await loadAuthorCollections()
+    const author = authorCollection.authors.find(author => author.id === authorId)
+
```

Comment:
Suggestion to make author non-optional in the props. 
```suggestion
    const author = authorCollection.authors.find(author => author.id === authorId)
    // Verify that the author exists. This normally shouldn't throw because `getStaticPaths` is supposed to only provide existing authors, but this check satisfies TypeScript that the author is never undefined.
    if (!author) {
        throw new Error(`Did not find author with id "${authorId}.`)
    }
```

Reply:
I guess with Lisa's suggestion, we can keep author optional, since we can have generic "Sourcegraph" authored articles too.

Reply:
After thinking about this, I think this is valuable feedback and is valid. An article can still be shown if it has no author but  when the author page is being accessed with an invalid author, the author page should not show if there is no author. I have updated the PR with this feedback.